app/bundles/CoreBundle/Resources/views/Helper/list_toolbar.html.twig

```diff
@@ -40,8 +48,191 @@
 
+        {# Existing toolbar actions #}
         {% include '@MauticCore/Helper/toolbar_actions.html.twig' %}
         {{ include('@MauticCore/Helper/toolbar_page_actions.html.twig', page_actions|default({})) }}
+
+        {# Added Columns button #}
+        {% if showColumnsButton %}
+        <button
+            type="button"
+            class="btn btn-default ml-2"
+            id="column-config-toggle"
+            data-toggle="modal"
+            data-target="#column-config-modal"
+            title="{{ 'mautic.core.columns'|trans }}"
+        >
+            <i class="fa fa-columns"></i> {{ 'Columns'|trans }}
+        </button>
+        {% endif %}
     </div>
 
 </div>
+
+<script>
+window.userColumnPrefs = {{ userColumnPrefs|json_encode|raw }};
+window.contactColPrefs = {{ contactColPrefs|json_encode|raw }};
+window.companyColPrefs = {{ companyColPrefs|json_encode|raw }};
+
+window.applyColumnPrefs = function(prefs, tableSelector, columnPrefix, allColumns) {
+  allColumns.forEach(col => {
+    const show = prefs.includes(col);
+    document.querySelectorAll(`${tableSelector} th.${columnPrefix}${col}, ${tableSelector} td.${columnPrefix}${col}`)
+      .forEach(el => el.style.setProperty('display', show ? '' : 'none', 'important'));
+  });
+};
+</script>
+
+{# Modal renderer #}
+<script>
+function renderColumnModal(config) {
+  const existing = document.getElementById('column-config-modal');
+  if (existing) existing.remove();
+
+  const formHtml = config.allColumns.map(key => {
+    const label = config.labels[key] || key;
+    const isChecked = (window.userColumnPrefs || []).includes(key);
+    return `
+      <div class="form-check">
+        <input class="form-check-input" type="checkbox" value="${key}" id="col-${key}" name="columns" ${isChecked ? 'checked' : ''}>
+        <label class="form-check-label" for="col-${key}">${label}</label>
+      </div>
+    `;
+  }).join('');
+
+  const modalHtml = `
+    <div class="modal" id="column-config-modal" tabindex="-1" role="dialog" aria-labelledby="columnConfigModalLabel" aria-hidden="true">
+      <div class="modal-dialog" role="document">
+        <div class="modal-content" tabindex="0">
+          <div class="modal-header">
+            <h5 class="modal-title" id="columnConfigModalLabel">Select Columns</h5>
+            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+              <span aria-hidden="true">&times;</span>
+            </button>
+          </div>
+          <div class="modal-body">
+            <form id="column-picker-form">${formHtml}</form>
+          </div>
+          <div class="modal-footer">
+            <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+            <button type="button" class="btn btn-primary" id="save-columns">Save changes</button>
+          </div>
+        </div>
+      </div>
+    </div>
+  `;
+
+  document.body.insertAdjacentHTML('beforeend', modalHtml);
+}
+</script>
+
+<script>
```

Comment:
Move this script to its file.

Reply:
@PedroLoureiro1 all scripts should be loaded from separate files, not combined with Twig or PHP etc.